tools/go-cli/go-whisk-cli/commands/commands.go

```diff
@@ -24,10 +24,47 @@ import (
 
     "../../go-whisk/whisk"
     "errors"
+    "github.com/spf13/cobra"
 )
 
 var client *whisk.Client
 
+func setupClientConfig(cmd *cobra.Command, args []string) (error){
+    var apiHostBaseUrl = fmt.Sprintf("https://%s/api/", Properties.APIHost)
+
+    baseURL, err := url.Parse(apiHostBaseUrl)
+
+    if err != nil {
+        whisk.Debug(whisk.DbgError, "url.Parse(%s) error: %s\n", apiHostBaseUrl, err)
+        errMsg := fmt.Sprintf("Invalid apiHost value '%s' : %s", Properties.APIHost, err)
+        whiskErr := whisk.MakeWskErrorFromWskError(errors.New(errMsg), err, whisk.EXITCODE_ERR_GENERAL,
+        whisk.DISPLAY_MSG, whisk.DISPLAY_USAGE)
+
+        return whiskErr
+    }
+
+    clientConfig := &whisk.Config{
+        AuthToken:  Properties.Auth,
+        Namespace:  Properties.Namespace,
+        BaseURL:    baseURL,
+        Version:    Properties.APIVersion,
+        Insecure:   flags.global.insecure,
+    }
+
+    // Setup client
+    client, err = whisk.NewClient(http.DefaultClient, clientConfig)
```

Comment:
@dubeejw - The init() method also initializes the client variable with a new NewClient object.  Is this still needed?